source/Nuke.Common/Tools/OctoVersion/OctoVersionTasks.cs

```diff
@@ -31,16 +40,18 @@ internal static string GetToolPath(string framework = null)
 
         private static OctoVersionInfo GetResult(IProcess process, OctoVersionGetVersionSettings toolSettings)
         {
+            ControlFlow.Assert(!string.IsNullOrEmpty(toolSettings.OutputJsonFile), $"{nameof(toolSettings.OutputJsonFile)} must be set");
```

Comment:
So just for clarification, the output file is now mandatory? I can't invoke OctoVersion and just let it set the environment variables?

Reply:
Yes.  
When moving from our inhouse attribute to this new built in approach, the execution method was also changed from executing OctoVersion as a c# library to executing it as an autogenerated Nuke CLI tool.  Executing it on CLI means we need to parse the output as Json to get the c# object exposed to the rest of the c# Nuke build script (eg the `OctoVersionInfo versionInfo`).  The initial implementation in this PR was running OctoVersion twice (once with Json output for parsing, then a 2nd time with auto detected CI output to set variables in the CI platform).  This new implementation runs it once, using an output file for the parseable json output and the console output for the autodetected CI platform.
The file is created in the NUKE temp area and is deleted after it's parsed.

Reply:
Understood. Did you also consider putting them all in env vars? Maybe even the whole json string? I'm just genuinely curious, because I also have a couple of files that are put into temp directory. So no objects to keep it as is.

Reply:
we did consider a few options but this was the best option given the way octoversion works, how the old (internal) Nuke.OctoVersion worked and not wanting to change too many things 🤣 We will stick with the file for now